### PR TITLE
 Added real name boolean option for the /list command 

### LINKED ISSUE #900: [Improvement] Add real name boolean option for the /list command
Being able to display player's Real name instead of or next to the DisplayName in the /list command output would be very appreciated :)

The following output (as an example) for each player when the displayRealNameInList value is set to true in the config would be welcomed:
`(playerRealName) playerDisplayName`
or
`playerDisplayName (playerRealName)`

### PR DESCRIPTION
**Pull request by Radoje17**
Fixes #900

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/ISettings.java [hunk 1]
```diff
@@ -3,7 +3,6 @@
 import com.earth2me.essentials.commands.IEssentialsCommand;
 import com.earth2me.essentials.signs.EssentialsSign;
 import com.earth2me.essentials.textreader.IText;
-
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/ISettings.java]: Why did you delete a line randomly here?

File: Essentials/src/com/earth2me/essentials/PlayerList.java [hunk 2]
```diff
@@ -26,7 +27,12 @@ public static String listUsers(final IEssentials ess, final List<User> users, fi
                 groupString.append(tl("listHiddenTag"));
             }
             user.setDisplayNick();
-            groupString.append(user.getDisplayName());
+
+            if (ess.getSettings().realNamesOnList() && !ChatColor.stripColor(user.getDisplayName()).equals(user.getName())) {
+                groupString.append(user.getDisplayName() + " (" + user.getName() + ")");
+            } else {
+                groupString.append(user.getDisplayName());
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/PlayerList.java]: Better: append this unconditionally, and only append the other part based on the condition.

File: Essentials/src/com/earth2me/essentials/Settings.java [hunk 3]
```diff
@@ -24,17 +24,8 @@
...
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Locale;
-import java.util.Map;
+import java.util.*;
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/Settings.java]: Please don't do this! I know the IDE will do this automatically, but you can turn it off. Wildcards are great in some cases, but I don't think this is one of them.

File: Essentials/src/config.yml [hunk 4]
```diff
@@ -381,6 +381,8 @@ list:
     # Uncomment the line below to simply list all players with no grouping
     #Players: '*'
 
+real-names-on-list: false
```
Comment (pop4959) [inline Essentials/src/config.yml:385]: Every config option should have a description. What's clear to you may not be for others.

**Review by pop4959 — CHANGES_REQUESTED** [review]
Appears to be functionally correct, but there are a few minor points to improve upon. It would also be nice to see a screenshot of how this option appears in-game (with both display names only, with username, and both mixed).

File: Essentials/src/com/earth2me/essentials/PlayerList.java (hunk 2 above)
Comment (pop4959) [reply Essentials/src/com/earth2me/essentials/PlayerList.java]: Oh also on second thought, you should also be using append instead of concatenation.

**Review by Radoje17 — COMMENTED** [review]
Screenshots:
image
image
image

**Review by Radoje17 — COMMENTED** [review]
Missed one parenthesis

File: Essentials/src/com/earth2me/essentials/PlayerList.java [hunk 5]
```diff
@@ -26,7 +27,13 @@ public static String listUsers(final IEssentials ess, final List<User> users, fi
                 groupString.append(tl("listHiddenTag"));
             }
             user.setDisplayNick();
+
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/PlayerList.java]: Looks like you added extra space here for whatever reason

File: Essentials/src/com/earth2me/essentials/PlayerList.java [hunk 6]
```diff
@@ -26,7 +27,13 @@ public static String listUsers(final IEssentials ess, final List<User> users, fi
...
             groupString.append(user.getDisplayName());
+            if (ess.getSettings().realNamesOnList() && !ChatColor.stripColor(user.getDisplayName()).equals(user.getName())) {
+                groupString.append(" (");
```
Comment (pop4959) [inline Essentials/src/com/earth2me/essentials/PlayerList.java]: Might look cleaner if you chain append calls (this is common for StringBuilder; example: `.append("a").append(b)` -> `"ab"`).

File: Essentials/src/config.yml [hunk 7]
```diff
@@ -381,6 +381,9 @@ list:
     # Uncomment the line below to simply list all players with no grouping
     #Players: '*'
 
+# Show realnames on /list if player has a nickname
```
Comment (pop4959) [inline Essentials/src/config.yml]: I think this should still be clarified a bit more. Maybe something like:

"Displays real names in /list next to players who are using a nickname."

**Review by pop4959 — CHANGES_REQUESTED** [review]
Thanks for giving some screenshots of your test cases, that looks good to me!

I still found a few minor things to nitpick on, but they aren't difficult to fix and then this PR will be perfect!

**Review by Radoje17 — COMMENTED** [review]
Fixed

**Review by pop4959 — APPROVED** [review]

**Pull request merged**